doc/cabal-project.rst

```diff
@@ -1432,6 +1432,24 @@ Advanced global configuration options
     the ``-package-env -`` option that allows ignoring the package
     environment files).
 
+.. cfg-field:: buildinfo: True, False
+               --enable-buildinfo
+               --disable-buildinfo
+    :synopsis: Whether build information for each individual component should be
+               written in a human readable format.
+
+    :default: ``False``
+
+    Enable generation of build information for Cabal components. Contains very detailed information
+    on how to build an individual component, such as compiler version, modules of a component and
+    how to compile the component.
+
+    The output format is in json, and the exact location can be discovered from ``plan.json``.
```

Comment:
Exactly where the path can be found in plan.json seems undocumented still?

Reply:
`plan.json` is undocumented in general, afaict. I did not really feel like documenting `plan.json`, too :) (But if that is what it takes to get this merged, so be it)

Reply:
I'd be happy with just an approximate description of where the build-info field is. Something like:

> the exact location is available in the `"build-info"` field of each build component in the `"install-plan"` map in ``plan.json``.

(didn't check if that's correct though)

Reply:
Also this should explicitly document that the field can be missing/null when the Cabal version too old.

Reply:
This still doesn't seem to mention what the field name in plan.json is or where that field can be found like I suggested above.